source/agora/consensus/Validation.d

```diff
@@ -956,6 +957,53 @@ public bool isValidEnrollment (const ref Enrollment enrollment,
     return isInvalidEnrollmentReason(enrollment, block_height, findUTXO) is null;
 }
 
+/*******************************************************************************
+
+    Check the validity of a pre-image information
+
+    Pre-image infomation is considered valid if:
+        - The keys for enrollment in two pieces of pre-image information are
+            same
+        - Height for a pre-image is greater than height for a current image
+        - A current image is same as n times hashed value of a pre-image
+            (n = difference of two heights).
+
+    Params:
+        pre_image = The pre-image information
+        cur_image = The current image information
```

Comment:
How about naming them `prev_image` and then `next_image` or `new_image`, just to make it obvious which preimage was already in the EnrollmentManager?

Reply:
> How about naming them prev_image and then next_image or new_image, just to make it obvious which preimage was already in the EnrollmentManager?

Good suggestion!